tests/listobject_test.py

```diff
@@ -1,32 +1,205 @@
+        print(listobjects_result)
+        self.assertEqual(len(listobjects_result['Contents']), 2, "Wrong number of results")
+        self.assert_key_in_contents_list(listobjects_result, 'dir1/dir1a/d1af1')
+        self.assert_key_in_contents_list(listobjects_result, 'dir1/dir1a/d1af2')
+
+    def test_botocore_list_with_delimiter_prefix_no_slash(self):
+        try:
+            # With a delimiter and not ending in a slash, this will return all keys beginning with the common
+            # prefix but will not descend into collections
+            assert_command('imkdir %s/commonkeyprefix_dir' % self.bucket_irods_path)
+            assert_command('iput f1 %s/commonkeyprefix_f1' % self.bucket_irods_path)
+            assert_command('iput f1 %s/commonkeyprefix_dir/f1' % self.bucket_irods_path)  # this one will not show up in this query
+
+            listobjects_result = self.client.list_objects_v2(Bucket=self.bucket_name, Delimiter='/', Prefix='commonkeyprefix')
+            print(listobjects_result)
+            self.assertEqual(len(listobjects_result['Contents']), 1, "Wrong number of results")
+            self.assert_key_in_contents_list(listobjects_result, 'commonkeyprefix_f1')
+            self.assertEqual(len(listobjects_result['CommonPrefixes']), 1, "Wrong number of results")
+            self.assert_prefix_in_common_prefixes_list(listobjects_result, 'commonkeyprefix_dir/')
+
+        finally:
+            # local cleanup
+            assert_command('irm -rf %s/commonkeyprefix_dir %s/commonkeyprefix_f1' % (self.bucket_irods_path, self.bucket_irods_path))
+
+    def test_botocore_list_no_delimiter(self):
+
+       # With no delimiter this will return all keys beginning with the common prefix and will descend into all collections
+
+       listobjects_result = self.client.list_objects_v2(Bucket=self.bucket_name, Prefix='di')
+       print(listobjects_result)
+       self.assertEqual(len(listobjects_result['Contents']), 6, "Wrong number of results")
+       self.assert_key_in_contents_list(listobjects_result, 'dir1/d1f1')
+       self.assert_key_in_contents_list(listobjects_result, 'dir1/d1f2')
+       self.assert_key_in_contents_list(listobjects_result, 'dir1/dir1a/d1af1')
+       self.assert_key_in_contents_list(listobjects_result, 'dir1/dir1a/d1af2')
+       self.assert_key_in_contents_list(listobjects_result, 'dir1/dir1b/d1bf1')
+       self.assert_key_in_contents_list(listobjects_result, 'dir1/dir1b/d1bf2')
+
+       # No common prefixes when there isn't a delimiter
+       self.assertRaises(KeyError, lambda: listobjects_result['CommonPrefixes'])
+
+    def test_botocore_list_nothing_found(self):
+       listobjects_result = self.client.list_objects_v2(Bucket=self.bucket_name, Prefix='doesnotexist')
+       print(listobjects_result)
+       self.assertRaises(KeyError, lambda: listobjects_result['Contents'])
+
+    def test_aws_list_with_delimiter_no_prefix(self):
```

Comment:
Consider grouping Boto tests and AWS tests into separate classes.

That gives you room to configure their environments differently without risk.